frontend/README.md

```diff
@@ -142,6 +142,45 @@ it('does something', () => {
 We use `jest`'s [`expect`](https://facebook.github.io/jest/docs/en/expect.html) assertion tool.
 
 
+## Localization
+
+The user interface is localized using [Fluent](https://projectfluent.org/) and the library `fluent-react`. Fluent allows to move the complexity of handling translated content from the developer to the translator. Thus, when using it, you should care only about the English version, and trust that the localizers will know what to do with your string in their language.
+
+### Adding localized content
+
+Every time you add a piece of UI that contains some content, there are 2 things you need to do. First, make sure you put it inside a `<Localized>` element, and give it an id. String identifiers should follow this form: `${module}-${component}-${contentDescription}`. For example, if you were to add a new button (`<button>Update</button>`) to the `Editor` component in the `entitydetails` module, the id for that button might be `entitydetails-editor-button-update`.
+
+That would give:
+
+```js
+class Editor extends React.Component {
+    render() {
+        return <div>
+            <Localized id="entitydetails-editor-button-update">
+                <button>Update</button>
+            </Localized>
+        </div>;
+    }
+}
+```
+
+The second thing you need to do is to add that new string into our main translation files. Translations available to localizers are based on those files, so if you forget to put the new content there, no one will be able to translate that content.
+
+The source language for our application is `en-US` (that is what you should use by default when writing content in the code, and in the language files). You can find them in `frontend/public/static/locale/en-US/`. We currently have only one file, named `translate.ftl`, and that is where all content should be added.
```

Comment:
I think it would be amazing to be able to have FTL files right next to the file containing the strings, like we do right now with `.css` and `.test.js` files. However, I don't see how that would work with translations collection (for Pontoon for example) and translations storage (because I don't want to have one file per locale in my code).

However, I can imagine some automation that would collect all `.ftl` file from the `src/` folder and merge them into one / some main files, in a locale folder, and that would serve as the translations reference.

Reply:
That's a good point. In the future, we might want to build tools to help with the collection step.

For now, keeping everything in a single file will make things much easier. We could consider splitting translations into smaller files before the single file gets _too_ long. I think it would require some kind of an `l10n.toml` project config, however? CC @Pike.

Reply:
Ah, nvm, I see that @Pike already wrote a lengthy comment about this.

Reply:
Filed [bug 1504710](https://bugzilla.mozilla.org/show_bug.cgi?id=1504710).